src/napari_threedee/manipulators/point_manipulator.py

```diff
@@ -1,201 +1,130 @@
 
     def _on_selection_change(self, event=None):
-        if self._layer._is_selecting is True or self.enabled is False:
+        print('selection changing!')
+        if self.layer is None:
             return
-
-        selected_points = list(self._layer.selected_data)
+        elif self.enabled is False:
+            return
+        elif self.layer._is_selecting is True:
+            return
+        selected_points = list(self.layer.selected_data)
         if len(selected_points) == 1:
             remove_mouse_callback_safe(
-                self._layer.mouse_drag_callbacks,
+                self.layer.mouse_drag_callbacks,
                 napari_selection_callback
             )
             add_mouse_callback_safe(
-                self._layer.mouse_drag_callbacks,
+                self.layer.mouse_drag_callbacks,
                 self.napari_selection_callback_passthrough
             )
             self.visible = True
-            self.translation = self.active_point_position
-
+            # if self._backend.is_dragging is False:
+            # toggling following line on/off breaks magnitude of translator vector
+            # but is needed to update position of manipulator when selection is changed...
+            self.origin = self.active_point_position
+            print('origin updated')
         else:
             self.visible = False
-            if self._layer.mode == 'select':
+            if self.layer.mode == 'select':
                 remove_mouse_callback_safe(
-                    self._layer.mouse_drag_callbacks,
+                    self.layer.mouse_drag_callbacks,
                     self.napari_selection_callback_passthrough
                 )
                 add_mouse_callback_safe(
-                    self._layer.mouse_drag_callbacks, napari_selection_callback
+                    self.layer.mouse_drag_callbacks, napari_selection_callback
                 )
 
-    def _pre_drag(
-            self,
-            click_point: np.ndarray,
-            selected_translator: Optional[int],
-            selected_rotator: Optional[int]
-    ):
-        pass
-
-    def _while_dragging_translator(self, selected_translator: int, translation_vector: np.ndarray):
-        self._layer._move([self.active_point_index], self.translation)
-        self._drag_start = None
+    def _pre_drag(self):
+        self.layer.events.highlight.disconnect(self._on_selection_change)
 
-    def _while_dragging_rotator(self, selected_rotator: int, rotation_matrix: np.ndarray):
-        # todo: store rotmat somewhere
-        pass
+    def _while_dragging_translator(self):
+        selected_point_index = list(self.layer.selected_data)[0]
+        self.layer.data[selected_point_index] = self.origin
+        # refresh rendering manually after modifying array data inplace
+        self.layer.refresh()
 
     def _post_drag(self):
-        pass
+        self.layer.events.highlight.connect(self._on_selection_change)
```

Comment:
please use the safe connect function

Reply:
as above, safe connect is for mouse callbacks